src/pages/ShoppingList.js

```diff
@@ -35,43 +64,85 @@ const ShoppingList = ({ token }) => {
     useEffect(() => {
         getShoppingList();
     }, []);
+
     const lessThan24Hours = date => {
         const formattedDate = parseInt(moment(date).format());
         const newDate = moment(Date.now());
         const lastPurchase = moment(formattedDate);
         return lastPurchase.diff(newDate, 'hours') < 24;
     };
 
-    const handleCheck = e => {
-        let db = fb.firestore();
-        let tokenRef = db.collection(userToken).doc(e.target.name);
-        let dataCheck = {
-            isChecked: e.target.checked,
-            lastPurchaseDate: moment(Date.now()).format(),
-        };
-        tokenRef.update(dataCheck).then(function() {
-            getShoppingList();
-        });
+    const HOURS24 = 86400; //24 hours in seconds
+
+    const handleCheck = (e,item) => {
+        if (!(item.lastPurchaseDate == null)) {
```

Comment:
```suggestion
        if (item.lastPurchaseDate) {
```
This will just check if the `item.lastPurchaseDate` exists (it'll go to the else statement if it's null).

Reply:
we tried to make it `item.lastPurchaseDate`, but I think because the field is always existing (whether with a date or null), it always went to the if statement. That's why we specified if it was null.

Reply:
Huh...OK. Maybe in the future you could change it to `if(item.lastPurchaseDate !== null)` ... just makes it easier to read, I think, but nbd now...it works and that's cool.